src/kernel/src/process/vm/mod.rs

```diff
@@ -594,6 +605,115 @@ impl MemoryManager {
         Ok(pages.into_raw().into())
     }
 
+    fn sys_batch_map(self: &Arc<Self>, td: &VThread, i: &SysIn) -> Result<SysOut, SysErr> {
+        let dmem_fd: i32 = i.args[0].try_into().unwrap();
+        let flags: MappingFlags = i.args[1].try_into().unwrap();
+        let operations: *const BatchMapOp = i.args[2].into();
+        let num_of_ops: i32 = i.args[3].try_into().unwrap();
+        let num_of_processed_ops: *mut i32 = i.args[4].into();
+
+        if flags.bits() & 0xe0bffb6f != 0 {
+            return Err(SysErr::Raw(EINVAL));
+        }
+
+        let slice_size = num_of_ops.try_into().ok().ok_or(SysErr::Raw(EINVAL))?;
+        let operations = unsafe { std::slice::from_raw_parts(operations, slice_size) };
+
+        let mut ret = Ok(0.into());
+        let mut processed = 0;
+        for op in operations {
+            // TODO: implement remaining logic when op != MapTypeProtect && some addr != 0
+
+            ret = match op.op.try_into() {
```

Comment:
There should be a ? here I think, then You don't have to wrap every variant in Ok(..)

Reply:
I think that'll behave differently, returning Err without breaking the loop and writing the number of processed entries

Reply:
Yeah, you're right. Oops. We could really use try_blocks here, but they're not stable :(